src/modules/app/containers/Helmet.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+
+import CityModel from '../../endpoint/models/CityModel'
+import React from 'react'
+import { CATEGORIES_ROUTE } from '../routes/categories'
+import CategoriesMapModel from '../../endpoint/models/CategoriesMapModel'
+import { connect } from 'react-redux'
+import EventModel from '../../endpoint/models/EventModel'
+import ReactHelmet from 'react-helmet'
+import { EVENTS_ROUTE } from '../routes/events'
+import { DISCLAIMER_ROUTE } from '../routes/disclaimer'
+import { EXTRAS_ROUTE } from '../routes/extras'
+import { SEARCH_ROUTE } from '../routes/search'
+import { MAIN_DISCLAIMER_ROUTE } from '../routes/mainDisclaimer'
+import ExtraModel from '../../endpoint/models/ExtraModel'
+import { translate } from 'react-i18next'
+import compose from 'lodash/fp/compose'
+import { LANDING_ROUTE } from '../routes/landing'
+
+import type { Location } from 'redux-first-router/dist/flow-types'
+
+type Props = {
+  location: Location,
+  categories: ?CategoriesMapModel,
+  cities: ?Array<CityModel>,
+  events: ?Array<EventModel>,
+  extras: ?Array<ExtraModel>,
+  t: string => string
+}
+
+export class Helmet extends React.Component<Props> {
+  getRoutePageTitle (): ?string {
+    const {location, categories, events, extras, t} = this.props
+    const currentRoute = location.type
+    const pathname = location.pathname
+
+    switch (currentRoute) {
+      case CATEGORIES_ROUTE:
+        if (categories) {
+          const category = categories.findCategoryByUrl(pathname)
+          if (category) {
+            if (category.id !== 0) {
+              return category.title
+            }
+          }
+        }
+        break
+      case EVENTS_ROUTE:
+        const eventId = location.payload.eventId
+        if (eventId) {
+          if (events) {
+            const event = events.find(event => event.id === eventId)
+            if (event) {
+              return event.title
+            }
+          }
+        } else {
+          return t('events')
+        }
+        break
+      case DISCLAIMER_ROUTE:
+        return t('disclaimer')
+      case EXTRAS_ROUTE:
+        const extraAlias = location.payload.extraAlias
+        if (extraAlias) {
+          if (extras) {
+            const extra = extras.find(extra => extra.alias === extraAlias)
+            if (extra) {
+              return extra.title
+            }
+          }
+        } else {
+          return t('extras')
+        }
+        break
+      case SEARCH_ROUTE:
+        return t('search')
+      case MAIN_DISCLAIMER_ROUTE:
+        return t('disclaimer')
+      case LANDING_ROUTE:
+        return t('landing')
+    }
```

Comment:
I don't like that we have another huge "switcher" here... Imo, the routes should decide on their title themselves.
Maybe we can expand the objects in /modules/app/routes/ for that purpose?
If we add an appropriate flow-type there, we should also be safe, that we haven't forgot to set a title to some route.
Also the tests wouldn't bloat a single file, like here Helmet.spec.js, but instead be spread to all of the routes...